src/modules/retransmitter/retransmitter.cpp

```diff
@@ -99,12 +99,24 @@ static void wait_for_msg() {
         }
     }
 }
-extern "C" void task_retransmitter_fn(void) {
 
-    ros_echronos::NodeHandle nh;
+ros_echronos::Publisher<std_msgs::Float64> * publishers[NUM_MSG];
+std_msgs::Float64 msg;
+struct messageAdapter serial;
+ros_echronos::NodeHandle nh;
+// void task_read_to_buffer_fn(void);
+// void task_publish_buffer_fn(void);
+
+union Data buf_reading;
+union Data buf_ready;
+union Data buf_sending;
+
+bool is_buffer_ready = false;
+bool is_buffer_sent = false;
+
+extern "C" void task_retransmitter_fn(void) {
 
     nh.init("retransmit_fn", "retransmit_fn", RTOS_INTERRUPT_EVENT_ID_CAN_RECEIVE_EVENT, 0);
-    ros_echronos::Publisher<std_msgs::Float64> * publishers[NUM_MSG];

```

Comment:
From memory I think this needed to be in the function as well because of the issue above, but worth testing

Reply:
I was getting an error, something along the lines of `this needs to be a modifiable lvalue`. I will look into this error message later.